### PR TITLE
[CS2113T-F12-2] AniChan

### PR DESCRIPTION
**Pull request by ChanJianHao**
AniChan is an all-rounded tool to effectively create and organise anime lists with viewing statistics, efficiency-focused features, and tools to improve anime-watching experience.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,34 +1,565 @@
...
+*   `AnimeData`: Provides data from the anime source file.
+*   `Storage`: Reads data from, and writes data to, the hard disk.
+
+<br/>
+
+Below is an overall sequence diagram to help illustrate the general program flow and how the different objects interact with each other.
+
+<br/>
+
+![Main Sequence Diagram](images/Overall-Sequence-Diagram.png) <br/>
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md]: image
The sequence diagram dosen't seem to load in the DG. Seems to happen for other components below...

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1,34 +1,565 @@
...
+
+<br/>
+
+## 4. Implementation
+This section introduces the specific implementation details and design consideration of some features in AniChan.
+
+<br/>
+
+### 4.1 Estimation Feature
+
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md:293]: This portion seems overly confusing to a new user like me... i think a flow diagram/class diagram would greatly help with explaining this implementation.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1,34 +1,565 @@
...
+![Estimate Command Final State Diagram](images/EstimateCommand-Final-State.png) <br/>
+*Figure 10: Estimate Command Final State*
+
+<br/>
+
+The sequence diagram presented below depicts the interaction between the components for running the `estimate` command, provided that the user has entered a valid command.
+
+<br/>
+
+![Estimate Command Sequence Diagram](images/EstimateCommand-Sequence-Diagram.png) <br/>
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md]: Capture
The diagram is too complicated, do consider splitting the diagram up to smaller portions. It is also difficult to read the words inside the diagram

Comment (Bryanbeh1998) [inline docs/DeveloperGuide.md]: Diagram is too complex. The words may also be a little too small for people to read.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1,34 +1,565 @@
...
+<br/>
+
+#### 4.1.2 Design Consideration
+
+This section shows some design considerations taken when implementing the estimate feature.
+
+<br/>
+
+##### Aspect: When should the program read the script file
+
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md:344]: Capture
While i feel the explanations here are well done, It could be too wordy and overly complicated. It should be easier to understand with a general flow diagram to justify your design choices.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1,34 +1,565 @@
...
+The user executes `watchlist -s 2` to change his active watchlist to the second watchlist ("NewAnime") in the list.
+
+![Watchlist Command State After Select Diagram](images/WatchlistCommand-After-Select.png) <br/>
+*Figure 19: Watchlist Command State After Select*
+
+<br/>
+
+The user executes `watchlist -d 2` to delete the second watchlist ("NewAnime") in the list.
+
+![Watchlist Command State After Delete Diagram](images/WatchlistCommand-After-Delete.png) <br/>
```
Comment (Bryanbeh1998) [inline docs/DeveloperGuide.md]: Words are too small in the sequence diagram. Hard for the user to read and follow

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Bryanbeh1998) [reply docs/DeveloperGuide.md]: Perhaps you can consider breaking it down further

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1,34 +1,565 @@
...
+*Figure 20: Watchlist Command Final State After Delete*
+
+<br/>
+
+The sequence diagram presented below depicts the interaction between the components for running the `watchlist` command, provided that the user has entered a valid command.
+
+<br/>
+
+![Watchlist Command Sequence Diagram](images/WatchlistCommand-Sequence-Diagram.png) <br/>
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md]: Capture

Should the Sequence diagram be split into different portions? It is difficult to understand what's going on...

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1,34 +1,565 @@
...
+**Step 6:** After all `Anime` objects in the page window have been retrieved, it will return the printable result back to `BrowseCommand#execute()`. At this point it will utilise `BrowseCommand#sortBrowseOption()` again to reset the `AnimeData` list to its original form if it has been altered.
+
+**Step 7:** Once `AnimeData` has been sorted, it will return the result back to `Main` for printing.
+
+Currently, this result is not very exciting as it is just based on the Anime ID which is assigned randomly by our data source, AniList.
+
+An example scenario would be browsing the 2nd page of a **sorted** list in ascending order.
+The only step that would change would be at Step 3, where it will perform sorting of `AnimeData` list. 
+
+![Browse Object Diagram 3](images/Browse-Sorted-State.png) <br/>
```
Comment (BenardoTang) [inline docs/DeveloperGuide.md:440]: I like that there is a diagram for almost after every step, well done!

**Review by BenardoTang — COMMENTED** [review]
Very well documented and explained for the most part. However, do take a look at some of the diagrams as they seem overly complicated and some don't load in the website. Good Job!!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Bryanbeh1998) [inline docs/DeveloperGuide.md]: The diagram does not load on the website. Perhaps you can check again

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1,34 +1,565 @@
...
+The sequence diagram presented below depicts the interaction between the components for running the `estimate` command, provided that the user has entered a valid command.
+
+<br/>
+
+![Estimate Command Sequence Diagram](images/EstimateCommand-Sequence-Diagram.png) <br/>
+*Figure 11. Sequence Diagram for EstimateCommand*
+
+<br/>
+
+#### 4.1.2 Design Consideration
```
Comment (Bryanbeh1998) [inline docs/DeveloperGuide.md]: This section although well documented, may seem a little too wordy

**Review by Bryanbeh1998 — COMMENTED** [review]
Overall a very well documented DG! Would look forward to trying out your product. Perhaps you can work on making the sequence diagram less complex. There is a part where it may seem a little too wordy as well. Great job!

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1,34 +1,565 @@
...
+<br/>&nbsp;5.3 [Testing](#53-testing)
+<br/>&nbsp;5.4 [DevOps](#54-devops)
 
-{Describe the target user profile}
+<br/>
 
-### Value proposition
+## 1. Introduction
+**AniChan** is a command-line application written in **Java 11**. It is written using the Object-Oriented Programming (OOP) paradigm which provides us with means to structure a software program into organized, reusable and reusable pieces of code that makes it good for future improvements and revisions.
+<br/>
```
Comment (skyaxe97) [inline docs/DeveloperGuide.md]: Should a more detailed introduction of the program be included here to introduce the functionality of the application?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -1,34 +1,565 @@
...
+*   `AnimeData`: Provides data from the anime source file.
+*   `Storage`: Reads data from, and writes data to, the hard disk.
+
+<br/>
+
+Below is an overall sequence diagram to help illustrate the general program flow and how the different objects interact with each other.
+
+<br/>
+
+![Main Sequence Diagram](images/Overall-Sequence-Diagram.png) <br/>
```
Comment (skyaxe97) [inline docs/DeveloperGuide.md]: image

Is there a missing name of method call from XYZParser to Command? And also, would it be better if the line from the activation bar of XYZParser to Command be lower on the activation bar? (It doesn't appear too distinct on the website)

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -1,34 +1,565 @@
...
+![Watchlist Command State After Delete Diagram](images/WatchlistCommand-After-Delete.png) <br/>
+*Figure 20: Watchlist Command Final State After Delete*
+
+<br/>
+
+The sequence diagram presented below depicts the interaction between the components for running the `watchlist` command, provided that the user has entered a valid command.
+
+<br/>
+
+![Watchlist Command Sequence Diagram](images/WatchlistCommand-Sequence-Diagram.png) <br/>
```
Comment (skyaxe97) [inline docs/DeveloperGuide.md]: image
Maybe consider using reference frames to split the diagram into different portions to make it more readable? It may consequently help to increase the font size of the wordings as it does get a little too small to read.

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -1,34 +1,565 @@
...
+Charlie
+Echo
+Gamma
+```
+
+If the 2nd page of the list was requested instead with the command `browse -p 2`. 
+`BrowseCommand#buildBrowseOutput()` will shift its page window down by 1 page as depicted in the diagram below.
+
+![Browse Object Diagram 2](images/Browse-Default-State2.png) <br/>
+*Figure 13: Browse Next Page Object Diagram*
```
Comment (skyaxe97) [inline docs/DeveloperGuide.md]: image
Good use of object diagrams to reflect the states.

**Review by skyaxe97 — COMMENTED** [review]
Looks good and detailed! Do look out for when the fonts in the diagrams get a little too tiny. Besides that, good work!

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -1,34 +1,814 @@
...
+
+The UI component consists of a `UI` class that handles all user input and system output. The UI is only dependent on the `Main` class and does not interact directly with other classes ensuring high cohesiveness and separation of roles.
+
+The `Ui` component listens for: 
+*   the execution of commands to print the result of the Command.
+*   any exceptions thrown to show an error message to the user, instead of a program termination.
+
+<br/>
+
+### 3.3 Parser Component
```
Comment (chewyang) [inline docs/DeveloperGuide.md:208]: image
Hi! this is a very clear diagram but should this arrow be a arrow with an empty head instead to show inheritance?

**Comment by jtankw3** [comment]
Hi team F12-2, please rename your PR with CS2113T instead.

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -1,34 +1,814 @@
...
+
+> :memo: The table shows the three ArrayList objects in the column with the bookmark id. When adding a new anime id into the bookmark, the bookmark will initialise the anime episode to be 0 together with an empty note object.
+
+**Step 4.5:** The user executes `bookmark -a 1` and `bookmark -a 410` to add anime id 1 and 410 to the bookmark.
+
+![Bookmark State After More Add Diagram](images/Bookmark-After-Step4.5.png) <br/>
+*Figure 21: Bookmark Entries with more Add*
+
+The following sequence diagram shows how the `Add Bookmark` operation works:
+
```
Comment (chewyang) [inline docs/DeveloperGuide.md:958]: image
Great use of colours to show the components, however should this diagram include sequences already introduced earlier? will showing the sequence after the execute() be clearer?

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -1,34 +1,814 @@
...
+	1. To Heart 2
+	2. Cowboy Bebop
+	3. InuYasha the Movie 2: The Castle Beyond the Looking Glass
+```
+
+**Step 6:** The user executes `bookmark -d 1` command to delete the bookmark entry at bookmark id: 1. `Bookmark#deleteAnimeBookmark()` will then remove the Bookmark index from the `Bookmark`.
+
+![Bookmark State After Delete Diagram](images/Bookmark-After-Step6.png) <br/>
+*Figure 23: Bookmark Entries After Delete*
+
```
Comment (chewyang) [inline docs/DeveloperGuide.md:975]: image

Great use of tables to illustrate the bookmarking implementation!

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -1,34 +1,920 @@
...
+<br/>&nbsp;5.2 [Logging](#52-logging)
+<br/>&nbsp;5.3 [Testing](#53-testing)
+<br/>&nbsp;5.4 [DevOps](#54-devops)
 
-{Describe the target user profile}
+<br/>
 
-### Value proposition
+## 1. Introduction
+**AniChan** is a command-line application written in **Java 11**. It is written using the Object-Oriented Programming (OOP) paradigm which provides us with means to structure a software program into organized, reusable and reusable pieces of code that makes it good for future improvements and revisions.
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: Would it be better if your team included a bit more details of what this application does/functions?

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -1,34 +1,814 @@
...
+
+| Approach | Pros | Cons  |
+| --- | --- | --- |
+| 1. Interactive browsing, users can `flip` pages and `pick` to view specific anime | - Fluid and seamless browsing session. <br/>- Very good usability. | - Would require `BrowseCommand` to get user input. |
+| 2. Static browsing, users specify the page they want to access. | - Completely decoupled from Ui component <br/>- Allows for browse to be more precise in finding what the user wants |  - Not as seamless as the first approach but still usable. |
+
+Though the 1st approach could have created a more authentic browsing feature it would not fit the requirements as well as the 
+2nd approach. The 2nd approach allows for more precise browsing of pages means that more experienced users are able to utilise the 
+tool quicker and to the same effect as the first approach. As a result, the 2nd approach was chosen as a better fit of the requirements 
+and in favour of having an application that is highly object-oriented.
```
Comment (chewyang) [inline docs/DeveloperGuide.md]: Good to see that all aspects for the browsing have been well thought out and considered before making your decision on the implementation to go for. Well documented as well!

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -1,34 +1,920 @@
...
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: Was the intention of the arrows from the main component to not be connected (touching) the other components? At the current moment, I feel a little unsure of where the main arrows should be pointing at.

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -1,34 +1,920 @@
...
+
+*Figure 1: Architecture Design Diagram*
+
+> :bulb: The images used are stored in the directory: `images/`. If you wish to update a diagram you may replace the images in this folder.
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: It might be good to indicate what the light bulb image means at the beginning of the DG to help the user understand the logos and images used in this document!

**Review by chewyang — COMMENTED** [review]
DG is well formatted and consistent, looks neat and diagrams are accurate and used when needed. Although some UML diagrams could be cropped and zoomed in to the use, omitting the main and parser classes as it may be a little repetitive. But overall great effort!

File: docs/DeveloperGuide.md [hunk 20]
```diff
@@ -1,34 +1,920 @@
...
+
+| Attribute | Option | Function          |
+| ---       | ---    | ---               |
+| order     | 0      | Ascending         |
+| order	    | 1      | Descending        |
+| sortType  | 0      | No Sort           |
+| sortType  | 1      | by name           |
+| sortType  | 2      | by rating         |
+| sortType  | 3      | back to original  |
+| page      | \>= 1  | page number       |
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: I like this concept of using a table as information is presented in a very readable and simple manner. The only question I have is with regards to the heading. Maybe it would be better if you explained to the user what the headings were before introducing them? If not it might lead to further confusion.

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -1,34 +1,920 @@
...
+object and return it to `main`.
+
+**Step 3:** Once the `BrowseCommand` object is executed, it will firstly process the `AnimeData` by calling on its `BrowseCommand#sortBrowseList()` operation to identify the order that the list will need to be in (if any).
+
+For this case since it is a default browse operation, there is no sorting performed.
+
+**Step 4:** Now `BrowseCommand` will utilise its `BrowseCommand#buildBrowseOutput()` operation to access all 
+`Anime objects within the page window, as shown in the diagram below.
+
+![Browse Object Diagram 1](images/Browse-Default-State.png) <br/>
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md:414]: image

Object diagrams do make it very simple for developers to understand the entire structure of a said component/function. From what I gathered, your team is explaining steps with respect to a given example. Would a sequence diagram be better at facilitating the flow of information since the users can read in a linear flow?

File: docs/DeveloperGuide.md [hunk 22]
```diff
@@ -1,34 +1,920 @@
...
+*Figure 12: Browse Sorted State Object Diagram*
+
+As you can see, even though the page window is at the same position as the previous command, 
+the list is different as it has been sorted.
+
+From this point onwards, the operation will continue as per the steps above but during the final `BrowseCommand#sortBrowseList()` call in step 6. It will perform a sort to reset the list.
+
+Here is the sequence diagram to better illustrate the lifecycle of a browse command.
+
+![Browse Sequence Diagram](images/Browse-SequenceDiagram.png) <br/>
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md:451]: image

Nice sequence diagram to illustrate the entire lifecycle of the browse command! I'm not sure if it's my device but there seems to be some missing arrows for return values? Or is that the intended course of action for the various methods that were called? Also, the activation bar should end at appropriate life cycles, however there are a few activation bars that are left elongated.

File: docs/DeveloperGuide.md [hunk 23]
```diff
@@ -1,34 +1,920 @@
...
+#### 4.3.1 Current Implementation
+The view information command is currently implemented by the `InfoCommand`. The user has to give an input of the form `info -a <ANIME_ID>`, and this would allow users to check all the information available for the ANIME_ID they have specified.
+
+Given below is an example of the usage scenario of view information command and how it behaves at each step.
+
+**Step 1:** `Ui` would receive input in the `Main` class and pass it into `Parser` class
+
+**Step 2:** The `Parser` class would then extract out `info` from the input given, which will instantiate a new `InfoParser` object, in which `InfoCommand` object is constructed as well.
+
+**Step3:** The `InfoParser#parse()` method will be invoked, and this will validate the parameters given by the user. Once validated, ANIME_ID will be set inside the `InfoCommand` object that was created previously. `InfoCommand` object will be returned back all the way to `Main`.
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: There might be an overlooked irregular spacing here!

File: docs/DeveloperGuide.md [hunk 24]
```diff
@@ -1,34 +1,920 @@
...
+![Workspace Command After Switch Diagram](images/WorkspaceCommand-After-Delete.png) <br/>
+*Figure 17: Workspace Command After New Workspace Delete*
+
+<br/>
+
+The following sequence diagram illustrates how `Workspace` creation in the example above works:
+
+> :memo: The other options (`-s`, `-l`, `-d`) follows a similar process, only the list and switch option does not interact with `StorageManager` and `Watchlist`.
+
+![Workspace Command Sequence Diagram](images/WorkspaceCommand-Sequence-Diagram.png) <br/>
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md:613]: image

Just to check, the box seems to be missing the label (eg. alt, opt)

File: docs/DeveloperGuide.md [hunk 25]
```diff
@@ -1,34 +1,920 @@
...
+
+#### 4.5.2 Design Considerations
+This section shows some design considerations taken when implementing the watchlist management features.
+
+Aspect: **Saving watchlist data**
+
+| Approach | Pros | Cons |
+| --- | --- | --- |
+| After each command execution **(current design)**. | User don't have to worry about lost data if their application or system crashes midway. | Application might slow down when the data grows large. |
+| When the user exits the program. | Saving is more efficient and could improve performance. | User may lose their data if the application or system crashes midway. |
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: Simple to read and understand! Good job 👍

File: docs/DeveloperGuide.md [hunk 26]
```diff
@@ -1,34 +1,920 @@
...
+
+**Step 4.5:** The user executes `bookmark -a 1` and `bookmark -a 410` to add anime id 1 and 410 to the bookmark.
+
+![Bookmark State After More Add Diagram](images/Bookmark-After-Step4.5.png) <br/>
+*Figure 26: Bookmark Entries with more Add*
+
+The following sequence diagram shows how the `Add Bookmark` operation works:
+
+![Bookmark Add Command Sequence Diagram](images/Bookmark-Add-Sequence-Diagram.png) <br/>
+*Figure 27: Bookmark Add Command Sequence Diagram*
```
Comment (yAOwzers) [inline docs/DeveloperGuide.md]: image

Good job on the sequence diagram, everything looks to be in-check! Easy to understand 👍

**Review by yAOwzers — COMMENTED** [review]
Overall good job for the DG! There are some formatting issues like irregular spacings between titles and headings, but other than that 🥇

**Comment by ChanJianHao** [comment]
> Hi team F12-2, please rename your PR with CS2113T instead.

Hello, we have renamed the PR. Thank you!

File: docs/DeveloperGuide.md [hunk 27]
```diff
@@ -1,34 +1,965 @@
...
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
```
Comment (MuhammadHoze) [inline docs/DeveloperGuide.md]: I suggest you could put main in the middle and connect the rest of the relevant boxes to it rather than the floating arrows which may arise confusion.

File: docs/DeveloperGuide.md [hunk 28]
```diff
@@ -1,34 +1,965 @@
...
+*   At shut down: Shuts down the components and invokes any clean up methods where necessary.
+
+The rest of AniChan consists of 6 components: 
+*   `Ui`: Manages the user interface of AniChan.
+*   `Parser`: Parses the user input.
+*   `Command`: Executes the command.
+*   `User`: Manages the workspace(s) and user data.
+*   `AnimeData`: Provides data from the anime source file.
+*   `Storage`: Reads data from, and writes data to, the hard disk.
+
```
Comment (MuhammadHoze) [inline docs/DeveloperGuide.md:176]: Nicely explained here what each class does.

File: docs/DeveloperGuide.md [hunk 29]
```diff
@@ -1,34 +1,965 @@
...
+**Step 1:** In the `Main` class, the user will input `view -v <WATCHLIST_ID>`. `Ui` will take in this input and is passed into `Parser`.
+
+**Step 2:** In `Parser’, `view` will be extracted out of the input, leading to a new `ViewWatchlistParser` object being instantiated, and in the constructor, a new `ViewWatchlistCommand` is created.
+
+**Step 3:** `ViewWatchListParser#parse()` is then called in `Parser`, which will validate the parameter that was given by the user. If the parameter is correct, the watchlist index will be set in the `ViewWatchlistCommand` object. The `ViewWatchlistCommand` object is then returned back to `Parser`, and back to `Main`.
+
+**Step 4:** The `ViewWatchlistCommand#execute()` would then be called by `Main`, in which the WATCHLIST_ID will be validated, and if valid, a string containing all the anime name inside the active `Watchlist` will be built and returned to `Main`, where it will be printed out by `Ui`.
+
+**Step 5:** `ViewWatchlistParser`, `ViewWatchlistCommand`, `Parser` and `Command` are terminated
+
```
Comment (MuhammadHoze) [inline docs/DeveloperGuide.md:889]: I suggest maybe having a diagram for certain steps here like for your other implementations for better understanding

File: docs/DeveloperGuide.md [hunk 30]
```diff
@@ -1,34 +1,965 @@
...
+Overview of the different features provided by bookmark
+The `bookmark` feature aims to provide the user with the ability to create shortcuts to anime they wish to track. This feature further provides tools to monitor the progress of a series and make informative notes.
+
+<br/>
+
+#### 4.9.1 Current Implementation
+
+The Bookmark class uses three ArrayList to store bookmark entries of the user, these arraylists maintain information about the anime index, episode and notes. The synchronisation between arraylist is required so that it enables easy retrieval of bookmark information using the bookmark index on the three arraylist.
+
+![Bookmark Class Diagram](images/Bookmark-Class-Diagram.png) <br/>
```
Comment (MuhammadHoze) [inline docs/DeveloperGuide.md:902]: Well done here! got a better picture on this implementation

File: docs/DeveloperGuide.md [hunk 31]
```diff
@@ -1,34 +1,965 @@
...
+The `Workspace` component:  
+*   can allow `User` to create and get the list of `Watchlist` and `Bookmark`.
+*   can allow `User` to change his active `Watchlist`.
+
+<!-- @@author -->
+
+<br/>
+
+### 3.7 Storage Component
+![Storage Class Diagram](images/Storage-Class-Diagram.png) <br/>
```
Comment (MuhammadHoze) [inline docs/DeveloperGuide.md]: Clear view of all the class diagram

**Review by MuhammadHoze — COMMENTED** [review]
Overall, good use of the tables and diagrams, maybe you could include them for most if not all of the implementations. Elaboration of each implementation is done clearly as well.

File: docs/DeveloperGuide.md (hunk 24 above)
Comment (ChanJianHao) [reply docs/DeveloperGuide.md:613]: Hello, may I know which box you're referring to?

File: docs/DeveloperGuide.md [hunk 32]
```diff
@@ -1,34 +1,1080 @@
...
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
```
Comment (homingjun) [inline docs/DeveloperGuide.md]: image
Where are the arrows from the main intended to point to? It might be better if the arrows point more specifically.

File: docs/DeveloperGuide.md [hunk 33]
```diff
@@ -1,34 +1,1080 @@
...
+<br/>
+
+### 3.2 UI Component
+![UI Class Diagram](images/Ui-Class-Diagram.png) <br/>
+*Figure 3: UI Class Diagram*
+
+The UI component consists of a `UI` class that handles all user input and system output. The UI is only dependent on the `Main` class and does not interact directly with other classes ensuring high cohesiveness and separation of roles.
+
+The `Ui` component listens for: 
+*   the execution of commands to print the result of the Command.
```
Comment (homingjun) [inline docs/DeveloperGuide.md]: image
Perhaps the Command could use the `Command` markdown instead since it is a major component, and also looks more consistent with the rest of the DG.

File: docs/DeveloperGuide.md [hunk 34]
```diff
@@ -1,34 +1,1080 @@
...
+![Main Sequence Diagram](images/Overall-Sequence-Diagram.png) <br/>
+*Figure 2: Overall Sequence Diagram*
+ 
+<br/>
+
+### 3.2 UI Component
+![UI Class Diagram](images/Ui-Class-Diagram.png) <br/>
+*Figure 3: UI Class Diagram*
+
+The UI component consists of a `UI` class that handles all user input and system output. The UI is only dependent on the `Main` class and does not interact directly with other classes ensuring high cohesiveness and separation of roles.
```
Comment (homingjun) [inline docs/DeveloperGuide.md]: image
It might be better UI could use the `UI` markdown as well to maintain the consistency of the markdowns.

File: docs/DeveloperGuide.md [hunk 35]
```diff
@@ -1,34 +1,1080 @@
...
+
+Aspect: **How should the program handle the sorted list**
+
+The first design consideration was how the sorting should be carried out. The main issues here are the time and storage complexity.
+
+| Approach | Pros | Cons  |
+| --- | --- | --- |
+| 1. Leaving the list unsorted                    | - No complexity and fastest approach  | - List will be unsorted and may cause confusion to users |
+| 2. Resorting the list again                     | - The list will be back into its original form before browsing    | - May hinder performance as resorting could take time <br/> - Requires altering of the main list |
+| 3. Cloning a duplicate animeData object to sort | - The list will be back to its original form <br/> - The main list will not be affected at all | - Expensive operation that will require large storage and time complexity |
```
Comment (homingjun) [inline docs/DeveloperGuide.md]: image
The use of a table is appropriate but it is a little wordy. Perhaps some icons could be used to increase the readability of the table.
